### ISSUE TITLE
Stabilize opentelemetry-semconv

### ISSUE THREAD

**Issue by jack-berg** [open-telemetry/semantic-conventions-java#90]
Want to start a conversation about what stands in the way of marking the `opentelemetry-semconv` artifact as stable. We've come a long way recently, re-organizing by domain, switching from build-tools to weaver based code generation, and changing how we think about deprecated / incubating attributes.

**Comment by breedx-splk** [comment]
With incubating in place, it seems like most of the roadblocks have been cleared. I suppose the main thing to think about is needing to be able to support future breaking changes though major semver bumps, which should be seldom and annoying at worst?

**Comment by jack-berg** [comment]
It would also be good to think through what might change in a future where we're generating helpers / constants for producing semantic convention metrics (#12), events, and anything else.

Does the class organization around root namespace work for this? For example, we currently have classes named `HttpAttributes` and `HttpIncubatingAttributes`. These names imply that the only hold the http attributes. If we added metric generation, would we feel comfortable adding classes like `HttpMetrics`, `HttpIncubatingMetrics`? 

Maybe we think of the current class organization as the embodiment of the semantic convention [attribute registry](https://github.com/open-telemetry/semantic-conventions/tree/main/docs/attributes-registry#attribute-registry) - a global registry of attributes agnostic of which conventions they're used in. Then, we can independently decide how to organize anything else we generate for _specific_ conventions.

**Comment by lmolkova** [comment]
I'd be really interested in stabilizing it for a selfish reason. I'd like to use [opentelemetry-instrumentation-api](https://mvnrepository.com/artifact/io.opentelemetry.instrumentation/opentelemetry-instrumentation-api) as a dependency in Azure SDK otel plugin, but it depends on `io.opentelemetry.semconv:opentelemetry-semconv` which is in alpha. Having alpha dependencies on this stable artifact gives me a pause - I would consider it a stability blocker for our plugin. It won't cause breaking changes on the API level, but I'd like to provide runtime behavior guarantees to my users.

It seems there are no major blockers preventing `io.opentelemetry.semconv:opentelemetry-semconv`  from being stabilized, so I wonder if it's something that can be considered in the near future.

**Comment by lmolkova** [comment]
I agree it'd be nice to make a final call on the code organization.

I'd like to call out a recent refactoring we've done around [semconv yaml folder structure](https://github.com/open-telemetry/semantic-conventions/tree/main/model) that'd be great to consider: 

```log
├── model
│   ├── {root-namespace}
│   │   ├── events.yaml
│   │   ├── metrics.yaml
│   │   ├── registry.yaml
│   │   ├── resources.yaml
│   │   ├── spans.yaml
```

https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md#code-structure

for Java, it'd probably translate to this:

```log
├── io/opentelemetry/semconv
│   ├── http
│   │   ├── HttpEvents.java   # if ever defined
│   │   ├── HttpMetrics.java
│   │   ├── HttpAttributes.java
│   │   ├── HttpSpans.java # if ever auto-generated
```

Following current pattern (attributes only), I'm not sure what would be the right place for metrics and other signals

```log
├── io/opentelemetry/semconv
│   ├── ExceptionEvents.java
│   ├── ExceptionMetrics.java # if ever..
│   ├── ExceptionAttributes.java
│   ├── HttpEvents.java   # if ever...
│   ├── HttpMetrics.java
│   ├── HttpAttributes.java
│   ├── HttpSpans.java # if ever..
│   ├── ......  # there will be a lot of files here
```

It feels like it needs some separation either by the root namespace or type (attribute/metric/span/event/resource). In semconv we decided to move away from type to the root namespace.

**Comment by trask** [comment]
I like this:

```
├── io/opentelemetry/semconv
│   ├── http
│   │   ├── HttpEvents.java   # if ever defined
│   │   ├── HttpMetrics.java
│   │   ├── HttpAttributes.java
│   │   ├── HttpSpans.java # if ever auto-generated
```

**Comment by trask** [comment]
(sorry @brunobat :sob:)

**Comment by jack-berg** [comment]
> I like this:

Do we gain enough from the additional directory to justify the churn? The role of packages is to:

- Organize code
- Prevent name collisions
- Interactions with class / method visibility keys

This reorg would is more aesthetically pleasing (who likes looking a a directory with hundreds of files?) but doesn't improve anything with respective to name collisions or visibility.

If we continue with the single package structure we have today, we still at least get all components of a particular namespace grouped when sorted.

**Comment by lmolkova** [comment]
It does not feel like the end of the world if we kept the current structure. If it was marked stable already, we'd not consider breaking it just for aesthetics.

**Comment by trask** [comment]
Discussed in Java SIG and decided to move forward with stabilizing the current structure.

TBD how to communicate that we are planning to drop the deprecated [`SemanticAttributes`](https://github.com/open-telemetry/semantic-conventions-java/blob/main/semconv/src/main/java/io/opentelemetry/semconv/SemanticAttributes.java) and [`ResourceAttributes`](https://github.com/open-telemetry/semantic-conventions-java/blob/main/semconv/src/main/java/io/opentelemetry/semconv/ResourceAttributes.java) classes.

**Comment by jack-berg** [comment]
Yes exactly. The RC has to be a preview of what to expect with the stable release.

**Comment by jack-berg** [comment]
I say we cut a release for 1.29.0 now, and target an RC / stable version for the next version of semantic conventions, 1.30.0. That gives us a concrete target for a stable version, without adding delay for anyone looking to use semantic-conventions:1.29.0, which was [released 3 weeks ago](https://github.com/open-telemetry/semantic-conventions/releases/tag/v1.29.0) already. 

Thoughts?

**Comment by trask** [comment]
works for me, I'll give the release workflow a try...

**Comment by jack-berg** [comment]
Completed in https://github.com/open-telemetry/semantic-conventions-java/releases/tag/v1.30.0
